File: engine/nexum-core/src/main/java/io/nexum/render/instructions/DrawRectInstruction.java
package io.nexum.render.instructions;

import io.nexum.models.Offset;
import io.nexum.models.Size;
import io.nexum.render.GraphicsInstruction;
import org.jetbrains.annotations.NotNull;

import java.awt.*;

public class DrawRectInstruction extends GraphicsInstruction {
    private final @NotNull Offset offset;
    private final @NotNull Size size;

    public DrawRectInstruction(@NotNull Offset offset, @NotNull Size size) {
        this.offset = offset;
        this.size = size;
    }

    @Override
    public void execute(@NotNull Graphics2D graphics) {
        graphics.drawRect(
                offset.getLeftPos(), offset.getTopPos(),
                size.getWidth(), size.getHeight()
        );
    }

    public @NotNull Offset getOffset() {
        return offset;
    }

    public @NotNull Size getSize() {
        return size;
    }
}

File: framework/nexum-core/lib/render/instructions/draw_rect_instruction.dart
import 'package:nexum_core/models/offset.dart';
import 'package:nexum_core/models/size.dart';
import 'package:nexum_core/render/graphics_instruction.dart';

class DrawRectInstruction extends GraphicsInstruction {
  final Offset offset;
  final Size size;

  DrawRectInstruction(this.offset, Size size) : size = Size(
      width: size.width - 1, height: size.height - 1
  );
}